### PR TITLE
Vulkan: 1.2.198.0 -> 1.3.211.0

### LINKED ISSUE #161060: vulkan-tools-lunarg fails to install with assertion error from vulkan-headers
### Describe the bug

`vulkan-tools-lunarg` fails to install with an assertion error from vulkan-headers

### Steps To Reproduce

1. Add the nixos-unstable channel
2. Run `nix-shell -p vulkan-tools-lunarg` --show-trace

### Expected behavior

vulkan-tools-lunarg installs normally

### Screenshots
```console
sky@home /mnt/hdd/projects/ALVR/packaging$ nix-shell -p vulkan-tools-lunarg --show-trace                       ✹v17
error: assertion '(version == (vulkan-headers).version)' failed

       at /nix/store/62fhzal89jx05gvaivj0zhy4w4g1qxjw-nixos-22.05pre353770.23d785aa6f8/nixos/pkgs/tools/graphics/vulkan-tools-lunarg/default.nix:28:10:

           27|
           28|   src = (assert version == vulkan-headers.version;
             |          ^
           29|     fetchFromGitHub {

       … while evaluating the attribute 'src' of the derivation 'vulkan-tools-lunarg-1.2.189.1'

       at /nix/store/62fhzal89jx05gvaivj0zhy4w4g1qxjw-nixos-22.05pre353770.23d785aa6f8/nixos/pkgs/stdenv/generic/make-derivation.nix:205:7:

          204|     // (lib.optionalAttrs (attrs ? name || (attrs ? pname && attrs ? version)) {
          205|       name =
             |       ^
          206|         let

       … while evaluating the attribute 'buildInputs' of the derivation 'shell'

       at /nix/store/62fhzal89jx05gvaivj0zhy4w4g1qxjw-nixos-22.05pre353770.23d785aa6f8/nixos/pkgs/stdenv/generic/make-derivation.nix:205:7:

          204|     // (lib.optionalAttrs (attrs ? name || (attrs ? pname && attrs ? version)) {
          205|       name =
             |       ^
          206|         let
```

### Additional context
Provavly the same issue as https://github.com/NixOS/nixpkgs/issues/156272

### Notify maintainers

@expipiplus1 

### Metadata
Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
sky@home /mnt/hdd/projects/ALVR/packaging$ nix-shell -p nix-info --run "nix-info -m"                                                     1 ↵  ✹v17
this path will be fetched (0.00 MiB download, 0.00 MiB unpacked):
  /nix/store/q124axggd6wl6fh9508sxgkknkifrl0z-nix-info
copying path '/nix/store/q124axggd6wl6fh9508sxgkknkifrl0z-nix-info' from 'https://cache.nixos.org'...
 - system: `"x86_64-linux"`
 - host os: `Linux 5.15.11-zen1, NixOS, 22.05 (Quokka)`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.5.1`
 - channels(root): `"home-manager, nixos-22.05pre353770.23d785aa6f8, stable-21.05.4719.0fd9ee1aa36"`
 - channels(sky): `"home-manager"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`
```

### PR DESCRIPTION
**Pull request by sgillespie**
Update:
 * glslang
 * spirv-headers
 * spirv-toools
 * vulkan-extension-layer
 * vulkan-headers
 * vulkan-loader
 * vulkan-tools
 * vulkan-tools-lunarg
 * vulkan-validation-layers

Fixes #161060

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Review by expipiplus1 — APPROVED** [review]

File: pkgs/tools/graphics/vulkan-extension-layer/default.nix
Comment (Ralith) [inline pkgs/tools/graphics/vulkan-extension-layer/default.nix:12]: What changed here?

Comment (expipiplus1) [reply pkgs/tools/graphics/vulkan-extension-layer/default.nix:12]: I assume it was 'invalidate the hash by replacing a character with `a`' but without replacing it with the correct hash (`sha256-ixCfHnp6YAOuR4NMTGjhhqkfm0H7ZcO/8xKFJqw16YE=`)

Comment (expipiplus1) [reply pkgs/tools/graphics/vulkan-extension-layer/default.nix:12]: @sgillespie Could you squash the commits and update this hash please?

**Comment by sgillespie** [comment]
I've updated all the hashes and squashed, but I'm still having one last issue with `vulkan-extension-layer`:

```
/nix/store/amq5z1ialh0qawfxbsn2sy768xi34mqb-vulkan-extension-layer-1.3.211.0/nix-support/setup-hook: line 2: XDG_DATA_DIRS: unbound variable
```

File: pkgs/tools/graphics/vulkan-extension-layer/default.nix
Comment (sgillespie) [reply pkgs/tools/graphics/vulkan-extension-layer/default.nix]: This is complete

**Review by Ralith — APPROVED** [review]

**Comment by yu-re-ka** [comment]
Broke wlroots build

**Comment by vcunat** [comment]
That's a channel blocker.  /cc also wlroots maintainers: @primeos, @Synthetica9.

**Comment by vcunat** [comment]
That's because this wrong path is in `vulkan.pc`

**Comment by expipiplus1** [comment]
Looks like 3c1ad4b0d54875ff0899b77a92aeda53ca236f27 in vulkan-loader is to blame

**Comment by expipiplus1** [comment]
Fix in https://github.com/NixOS/nixpkgs/pull/170748

**Comment by Mindavi** [comment]
error: Function called without required argument "pkgconfig" at /home/rick/.cache/nixpkgs-review/pr-171147/nixpkgs/pkgs/tools/graphics/vulkan-tools/default.nix:5, did you mean "pkg-config" or "pkgconf"?

also related?

**Comment by sgillespie** [comment]
> Since this got merged wlroots is not building for me. It complains about not being able to find "vulkan/vulkan.h"

I got this initially after merging as well, but I can now build against master following #170748. Still broken for you?

**Comment by sgillespie** [comment]
> error: Function called without required argument "pkgconfig" at /home/rick/.cache/nixpkgs-review/pr-171147/nixpkgs/pkgs/tools/graphics/vulkan-tools/default.nix:5, did you mean "pkg-config" or "pkgconf"?
> 
> also related?

I think it is--I think I should have used `pkg-config` rather than `pkgconfig`, even though it appears to work for me. Are you still having this issue?

**Comment by Mindavi** [comment]
It was something I saw in a seemingly unrelated PR, I can try running nixpkgs-review again and see if anything comes up still

**Comment by Mindavi** [comment]
Yep, still happening. Was running `nixpkgs-review pr 171147`. Probably still need to use pkg-config here since pkgconfig is an alias?

**Comment by Mindavi** [comment]
Hmm, also seeing this for cross-compilation just now:

<details><summary>Build log</summary>

```
unpacking sources
unpacking source archive /nix/store/qk7x5x78j2p943d4fs2l5l34hdyiwgk2-source
source root is source
patching sources
applying patch /nix/store/wyjhd12vda66y64x6m3q3i056k8z66vp-fix-pkgconfig.patch
patching file loader/vulkan.pc.in
updateAutotoolsGnuConfigScriptsPhase
configuring
fixing cmake files...
cmake flags: -DCMAKE_FIND_USE_SYSTEM_PACKAGE_REGISTRY=OFF -DCMAKE_FIND_USE_PACKAGE_REGISTRY=OFF -DCMAKE_EXPORT_NO_PACKAGE_REGISTRY=ON -DCMAKE_BUILD_TYPE=Release -DCMAKE_SKIP_BUILD_RPATH=ON -DBUILD_TESTING=OFF -DCMAKE_INSTALL_LOCALEDIR=/nix/store/ljxpsmnzi8h9f02nh0byxvis2ql3cyys-vulkan-loader-aarch64-unknown-linux-gnu-1.3.211.0/share/locale -DCMAKE_INSTALL_LIBEXECDIR=/nix/store/ljxpsmnzi8h9f02nh0byxvis2ql3cyys-vulkan-loader-aarch64-unknown-linux-gnu-1.3.211.0/libexec -DCMAKE_INSTALL_LIBDIR=/nix/store/ljxpsmnzi8h9f02nh0byxvis2ql3cyys-vulkan-loader-aarch64-unknown-linux-gnu-1.3.211.0/lib -DCMAKE_INSTALL_DOCDIR=/nix/store/ljxpsmnzi8h9f02nh0byxvis2ql3cyys-vulkan-loader-aarch64-unknown-linux-gnu-1.3.211.0/share/doc/Vulkan-Loader -DCMAKE_INSTALL_INFODIR=/nix/store/ljxpsmnzi8h9f02nh0byxvis2ql3cyys-vulkan-loader-aarch64-unknown-linux-gnu-1.3.211.0/share/info -DCMAKE_INSTALL_MANDIR=/nix/store/ljxpsmnzi8h9f02nh0byxvis2ql3cyys-vulkan-loader-aarch64-unknown-linux-gnu-1.3.211.0/share/man -DCMAKE_INSTALL_OLDINCLUDEDIR=/nix/store/q246rpxhmjg530bqs5nyclpyg1kl32qf-vulkan-loader-aarch64-unknown-linux-gnu-1.3.211.0-dev/include -DCMAKE_INSTALL_INCLUDEDIR=/nix/store/q246rpxhmjg530bqs5nyclpyg1kl32qf-vulkan-loader-aarch64-unknown-linux-gnu-1.3.211.0-dev/include -DCMAKE_INSTALL_SBINDIR=/nix/store/ljxpsmnzi8h9f02nh0byxvis2ql3cyys-vulkan-loader-aarch64-unknown-linux-gnu-1.3.211.0/sbin -DCMAKE_INSTALL_BINDIR=/nix/store/ljxpsmnzi8h9f02nh0byxvis2ql3cyys-vulkan-loader-aarch64-unknown-linux-gnu-1.3.211.0/bin -DCMAKE_INSTALL_NAME_DIR=/nix/store/ljxpsmnzi8h9f02nh0byxvis2ql3cyys-vulkan-loader-aarch64-unknown-linux-gnu-1.3.211.0/lib -DCMAKE_POLICY_DEFAULT_CMP0025=NEW -DCMAKE_OSX_SYSROOT= -DCMAKE_FIND_FRAMEWORK=LAST -DCMAKE_STRIP=/nix/store/aywldli8wshw2wfx0ryx14rq8ljlvfwk-aarch64-unknown-linux-gnu-stage-final-gcc-debug-wrapper-9.3.0/bin/aarch64-unknown-linux-gnu-strip -DCMAKE_RANLIB=/nix/store/rnhh2n9bmabrjm61sa2w4i3pza2lgadc-aarch64-unknown-linux-gnu-binutils-2.38/bin/aarch64-unknown-linux-gnu-ranlib -DCMAKE_AR=/nix/store/rnhh2n9bmabrjm61sa2w4i3pza2lgadc-aarch64-unknown-linux-gnu-binutils-2.38/bin/aarch64-unknown-linux-gnu-ar -DCMAKE_C_COMPILER=aarch64-unknown-linux-gnu-gcc -DCMAKE_CXX_COMPILER=aarch64-unknown-linux-gnu-g++ -DCMAKE_INSTALL_PREFIX=/nix/store/ljxpsmnzi8h9f02nh0byxvis2ql3cyys-vulkan-loader-aarch64-unknown-linux-gnu-1.3.211.0 -DCMAKE_INSTALL_INCLUDEDIR=/nix/store/izzs5b3l9g5s4xyvhdh21w63b8wsnnj2-vulkan-headers-aarch64-unknown-linux-gnu-1.3.211.0/include -DSYSCONFDIR=/run/opengl-driver/share -DCMAKE_SYSTEM_NAME=Linux -DCMAKE_SYSTEM_PROCESSOR=aarch64 -DCMAKE_HOST_SYSTEM_NAME=Linux -DCMAKE_HOST_SYSTEM_PROCESSOR=x86_64 
-- The C compiler identification is GNU 9.3.0
-- The CXX compiler identification is GNU 9.3.0
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /nix/store/aywldli8wshw2wfx0ryx14rq8ljlvfwk-aarch64-unknown-linux-gnu-stage-final-gcc-debug-wrapper-9.3.0/bin/aarch64-unknown-linux-gnu-gcc - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /nix/store/aywldli8wshw2wfx0ryx14rq8ljlvfwk-aarch64-unknown-linux-gnu-stage-final-gcc-debug-wrapper-9.3.0/bin/aarch64-unknown-linux-gnu-g++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
********************************************************************************
* NOTE: Not adding target to run update_deps.py automatically.                 *
********************************************************************************
-- Looking for pthread.h
-- Looking for pthread.h - found
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD - Success
-- Found Threads: TRUE  
-- Found VulkanHeaders: /nix/store/izzs5b3l9g5s4xyvhdh21w63b8wsnnj2-vulkan-headers-aarch64-unknown-linux-gnu-1.3.211.0/include  
-- Found VulkanRegistry: /nix/store/izzs5b3l9g5s4xyvhdh21w63b8wsnnj2-vulkan-headers-aarch64-unknown-linux-gnu-1.3.211.0/share/vulkan/registry  
-- Detected Vulkan Version 1.3.211
-- Found PkgConfig: /nix/store/f557q3qcn2zqyvbcs8xdaspgxm6y3j0w-aarch64-unknown-linux-gnu-pkg-config-wrapper-0.29.2/bin/aarch64-unknown-linux-gnu-pkg-config (found version "0.29.2") 
-- Could NOT find Git (missing: GIT_EXECUTABLE) 
Package xau was not found in the pkg-config search path.
Perhaps you should add the directory containing `xau.pc'
to the PKG_CONFIG_PATH environment variable
Package 'xau', required by 'xcb', not found
Package xau was not found in the pkg-config search path.
Perhaps you should add the directory containing `xau.pc'
to the PKG_CONFIG_PATH environment variable
Package 'xau', required by 'xcb', not found
Package xau was not found in the pkg-config search path.
Perhaps you should add the directory containing `xau.pc'
to the PKG_CONFIG_PATH environment variable
Package 'xau', required by 'xcb', not found
Package xau was not found in the pkg-config search path.
Perhaps you should add the directory containing `xau.pc'
to the PKG_CONFIG_PATH environment variable
Package 'xau', required by 'xcb', not found
-- Found xcb: /nix/store/4hcphzrzz1hk4yzqdn12l4spadhmqprs-libxcb-aarch64-unknown-linux-gnu-1.14-dev/include  
-- Found X11: /nix/store/w3q556a6rixxr25ix8zvawqvwbjydhnc-xorgproto-aarch64-unknown-linux-gnu-2021.5/include   
-- Looking for XOpenDisplay in /nix/store/1zgl9zm9ikna1xkaljfnc7nd305mmk30-libX11-aarch64-unknown-linux-gnu-1.7.2/lib/libX11.so
-- Looking for XOpenDisplay in /nix/store/1zgl9zm9ikna1xkaljfnc7nd305mmk30-libX11-aarch64-unknown-linux-gnu-1.7.2/lib/libX11.so - found
-- Looking for gethostbyname
-- Looking for gethostbyname - found
-- Looking for connect
-- Looking for connect - found
-- Looking for remove
-- Looking for remove - found
-- Looking for shmat
-- Looking for shmat - found
Package libffi was not found in the pkg-config search path.
Perhaps you should add the directory containing `libffi.pc'
to the PKG_CONFIG_PATH environment variable
Package 'libffi', required by 'wayland-client', not found
Package libffi was not found in the pkg-config search path.
Perhaps you should add the directory containing `libffi.pc'
to the PKG_CONFIG_PATH environment variable
Package 'libffi', required by 'wayland-client', not found
Package libffi was not found in the pkg-config search path.
Perhaps you should add the directory containing `libffi.pc'
to the PKG_CONFIG_PATH environment variable
Package 'libffi', required by 'wayland-client', not found
Package libffi was not found in the pkg-config search path.
Perhaps you should add the directory containing `libffi.pc'
to the PKG_CONFIG_PATH environment variable
Package 'libffi', required by 'wayland-client', not found
-- Found WAYLAND_CLIENT: /nix/store/2pnbgkcpc8q7m7c9k0519nxrgcmk3lwd-wayland-aarch64-unknown-linux-gnu-1.20.0/lib/libwayland-client.so  
-- Found WAYLAND_SERVER: /nix/store/2pnbgkcpc8q7m7c9k0519nxrgcmk3lwd-wayland-aarch64-unknown-linux-gnu-1.20.0/lib/libwayland-server.so  
-- Found WAYLAND_EGL: /nix/store/2pnbgkcpc8q7m7c9k0519nxrgcmk3lwd-wayland-aarch64-unknown-linux-gnu-1.20.0/lib/libwayland-egl.so  
-- Found WAYLAND_CURSOR: /nix/store/2pnbgkcpc8q7m7c9k0519nxrgcmk3lwd-wayland-aarch64-unknown-linux-gnu-1.20.0/lib/libwayland-cursor.so  
-- Found WAYLAND: /nix/store/2pnbgkcpc8q7m7c9k0519nxrgcmk3lwd-wayland-aarch64-unknown-linux-gnu-1.20.0/lib/libwayland-client.so;/nix/store/2pnbgkcpc8q7m7c9k0519nxrgcmk3lwd-wayland-aarch64-unknown-linux-gnu-1.20.0/lib/libwayland-server.so;/nix/store/2pnbgkcpc8q7m7c9k0519nxrgcmk3lwd-wayland-aarch64-unknown-linux-gnu-1.20.0/lib/libwayland-egl.so;/nix/store/2pnbgkcpc8q7m7c9k0519nxrgcmk3lwd-wayland-aarch64-unknown-linux-gnu-1.20.0/lib/libwayland-cursor.so  
-- Looking for secure_getenv
-- Looking for secure_getenv - found
-- Looking for __secure_getenv
-- Looking for __secure_getenv - not found
WARNING: VulkanLoader_generated_source target requires python 3
-- The ASM compiler identification is GNU
-- Found assembler: /nix/store/aywldli8wshw2wfx0ryx14rq8ljlvfwk-aarch64-unknown-linux-gnu-stage-final-gcc-debug-wrapper-9.3.0/bin/aarch64-unknown-linux-gnu-gcc
-- Configuring done
-- Generating done
CMake Warning:
  Manually-specified variables were not used by the project:

    BUILD_TESTING
    CMAKE_EXPORT_NO_PACKAGE_REGISTRY
    CMAKE_POLICY_DEFAULT_CMP0025


-- Build files have been written to: /build/source/build
cmake: enabled parallel building
building
build flags: -j8 -l8 SHELL=/nix/store/07ln9bxp9k8nds669r24fsywf4d1jlly-bash-5.1-p16/bin/bash
[  5%] Building C object loader/CMakeFiles/asm_offset.dir/asm_offset.c.o
[ 10%] Linking C executable asm_offset
[ 10%] Built target asm_offset
[ 15%] Generating gen_defines.asm
/nix/store/07ln9bxp9k8nds669r24fsywf4d1jlly-bash-5.1-p16/bin/bash: line 1: asm_offset: command not found
make[2]: *** [loader/CMakeFiles/loader_asm_gen_files.dir/build.make:73: loader/gen_defines.asm] Error 127
make[1]: *** [CMakeFiles/Makefile2:154: loader/CMakeFiles/loader_asm_gen_files.dir/all] Error 2
make: *** [Makefile:136: all] Error 2
```

</details>

Probably an upstream issue, but just wanted to note it here for reference anyway. AFAICT it seems to build asm_offset for the host platform, while it should be built for the build platform.

**Comment by Mindavi** [comment]
Meh, indeed an upstream issue.

**Comment by sgillespie** [comment]
> error: Function called without required argument "pkgconfig" at /home/rick/.cache/nixpkgs-review/pr-171147/nixpkgs/pkgs/tools/graphics/vulkan-tools/default.nix:5, did you mean "pkg-config" or "pkgconf"?

I've made the changes here: https://github.com/NixOS/nixpkgs/pull/171474

**Pull request merged**
